Articles/Section-Trees/Searchable-Trees/index.md

```diff
@@ -0,0 +1,139 @@
+# Searchable Trees (ISearchableTree)
+
+When you type a search term into the Umbraco backoffice search field, you'll see search results from all the Section Trees that your user account has permissions to access:
+
+![Content Section Dashboards](images/backoffice-search-v8.png)
+
+The results are grouped by 'Section Tree' e.g. Content, Media, Document Types: essentially, each 'Tree' has its own associated search mechanism that receives the search term and looks for matches in the tree that is responsible for searching.
+
+You can create your own search mechanisms for your own custom sections or replace the default search implementation for a particular section tree.
+
+## Custom Tree Search
+
+To create a search for your own custom tree you need to create a C# class that implements the interface `Umbraco.Cms.Core.Trees.ISearchableTree`.
+
+### ISearchableTree
+
+```csharp
+public interface ISearchableTree : IDiscoverable
+{
+    /// <summary>
+    /// The alias of the tree that the <see cref="ISearchableTree"/> belongs to
+    /// </summary>
+    string TreeAlias { get; }
+
+    /// <summary>
+    /// Searches for results based on the entity type
+    /// </summary>
+    /// <param name="query"></param>
```

Comment:
I'm a little unsure as to what you mean here, @AndyButland 🤔 
Are you saying we should **remove** all the comments here - so lines 20-35, or are you suggesting to include it, instead of it just being comments.
Or perhaps something completely different? 😅

Reply:
Sorry, I wasn't very clear... these comments are simply copied from source code and are there to explain to developers using these methods what the parameters you pass in are used for.  So they are useful to have in documentation.

The point I'm making though is that not all of them are complete - e.g. `<param name="query"></param>` should be something like `<param name="query">Search term used for finding matching results.</param>`.  So ideally would be good if they were completed, or if not, they aren't adding any value so could be removed.

Reply:
Ah! Thanks for explaining - makes a lot of sense actually completed the sample comments then 🤔 

Not entirely sure I have enough knowledge about this to do it though.. - would you be up for adding some text to the unfinished ones, @AndyButland ?

Reply:
Sure, doesn't look like I can edit this branch directly (probably as it's in a fork), but I can comment here.

For the those associated with this comment thread I'd suggest:

```
/// <param name="query">The search term used for finding matching results.</param>
/// <param name="pageSize">The number of records to return for a page of results.</param>
/// <param name="pageIndex">The 0-based index for retrieving a page of search results.</param>
/// <param name="totalFound">Populated with the total number of results matching the provided search term.</param>
/// <param name="searchFrom">The starting point for the search, generally a node ID, but for members this is a member type alias.</param>
```

In other files we have this comment in ` Articles/Section-Trees/sections.md`:

```
/// <summary>Compose.</summary>
```

Which I'd suggest simply removing, as it's not adding any information.

The rest I think is fine.

Reply:
Thanks @AndyButland !
I'll make sure to update the two files accordingly 💪